internal/pkg/agent/transpiler/vars_test.go

```diff
@@ -260,7 +260,7 @@ func TestVars_Replace(t *testing.T) {
 			if test.Error {
 				assert.Error(t, err)
 			} else if test.NoMatch {
-				assert.Error(t, ErrNoMatch, err)
+				assert.ErrorIs(t, err, ErrNoMatch)
 			} else {
 				require.NoError(t, err)
 				assert.Equal(t, test.Result, res)
```

Comment:
Nit: this whole block would probably be easier to read as an [ErrorAssertionFunc](https://pkg.go.dev/github.com/stretchr/testify/assert#ErrorAssertionFunc) member in the testcase struct, so input and error assertions can be kept together instead of signaling with a boolean